src/snowflake/cli/api/commands/snow_typer.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import logging
+from functools import wraps
+from typing import Optional
+
+import typer
+from snowflake.cli.api.commands.decorators import (
+    global_options,
+    global_options_with_connection,
+)
+from snowflake.cli.api.commands.flags import DEFAULT_CONTEXT_SETTINGS
+from snowflake.cli.api.exceptions import CommandReturnTypeError
+from snowflake.cli.api.output.types import CommandResult
+from snowflake.cli.app.printing import print_result
+from snowflake.cli.app.telemetry import flush_telemetry, log_command_usage
+
+log = logging.getLogger(__name__)
+
+
+class SnowTyper(typer.Typer):
+    def __init__(self, /, **kwargs):
+        super().__init__(
+            **kwargs,
+            context_settings=DEFAULT_CONTEXT_SETTINGS,
+            pretty_exceptions_show_locals=False,
+        )
+
+    @wraps(typer.Typer.command)
+    def command(
+        self,
+        name: Optional[str] = None,
+        requires_global_options: bool = True,
+        requires_connection: bool = False,
+        **kwargs,
+    ):
+        """
+        Custom implementation of Typer.command that adds ability to execute additional
+        logic before and after execution as well as process the result and act on possible
+        errors.
+        """
+        cls_super = super()
+
+        def custom_command(command_callable):
+            """Custom command wrapper similar to Typer.command."""
+            if requires_connection:
+                command_callable = global_options_with_connection(command_callable)
+            elif requires_global_options:
+                command_callable = global_options(command_callable)
+
+            @wraps(command_callable)
+            def command_callable_decorator(*args, **kw):
+                """Wrapper around command callable. This is what happens at "runtime"."""
+                self.pre_execute_callback()
+                try:
+                    result = command_callable(*args, **kw)
+                    return self.process_result(result)
+                except Exception as err:
+                    self.exception_execute_callback(err)
+                    raise
+                finally:
+                    self.post_execute_callback()
+
+            return cls_super.command(name=name, **kwargs)(command_callable_decorator)
```

Comment:
Is there any reason not to invoke `super` in regular way?

```suggestion
            return super(SnowTyper, self).command(name=name, **kwargs)(command_callable_decorator)
```

Reply:
Good one! I totally forgot about this syntax